### PR TITLE
Allow EVM to EWasm translation in assembly mode.

### PR DESCRIPTION
**Pull request by chriseth**
Feature required for testing in https://github.com/ethereum/solidity/pull/7766

Allow EVM to EWasm translation if `--assembly` mode. Only for commandline interface for now.

### CODE REVIEW

File: solc/CommandLineInterface.cpp [hunk 1]
```diff
@@ -1351,6 +1384,9 @@ bool CommandLineInterface::assemble(
 
 	for (auto const& src: m_sourceCodes)
 	{
+		// TODO translate from EVM to eWasm if
```
Comment (chriseth) [inline solc/CommandLineInterface.cpp]: Maybe better to do that in assembly stack directly?

Comment (axic) [reply solc/CommandLineInterface.cpp]: Yes, I'd prefer to do it over there.

File: solc/CommandLineInterface.cpp [hunk 2]
```diff
@@ -911,7 +924,27 @@ bool CommandLineInterface::processInput()
...
+				inputLanguage = Input::EWasm;
+				if (targetMachine != Machine::eWasm)
+				{
+					serr() << "If you select eWasm as --yul-dialect, --machine has to be eWasm as well." << endl;
+					return false;
+				}
+			}
+			else
+			{
+				serr() << "Invalid option for --yul-dialct: " << dialect << endl;
```
Comment (axic) [inline solc/CommandLineInterface.cpp]: Typo `yul-dialct`

**Review by leonardoalt — CHANGES_REQUESTED** [review]
Can you add some tests?

**Comment by chriseth** [comment]
The tests would have to be adapted all the time, but I can try.

**Comment by chriseth** [comment]
Added a test. It's actually quite nice!

**Comment by leonardoalt** [comment]
Weird leak test failure, rerunning it

File: solc/CommandLineInterface.cpp [hunk 3]
```diff
@@ -1331,14 +1364,14 @@ bool CommandLineInterface::assemble(
...
 		auto const& stack = sourceAndStack.second;
-		unique_ptr<SourceReferenceFormatter> formatter;
-		if (m_args.count(g_argNewReporter))
-			formatter = make_unique<SourceReferenceFormatterHuman>(serr(false), m_coloredOutput);
-		else
-			formatter = make_unique<SourceReferenceFormatter>(serr(false));
-
 		for (auto const& error: stack.errors())
 		{
+			unique_ptr<SourceReferenceFormatter> formatter;
```
Comment (axic) [inline solc/CommandLineInterface.cpp:1369]: Why create a new formatter for each error?

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**